control.lua

```diff
@@ -1,108 +1,103 @@
 script.on_init(
   function()
-    log("On init")
-    if not storage.train_stop_list then
-      log("Initializing train_stop_list")
-      storage.train_stop_list = {}
-    end
-    log("Train stop count: " .. #storage.train_stop_list)
+    storage.dynamicTrainStopSettings = {}
   end
 )
 
-script.on_event(defines.events.on_tick,
-  function(event)
-    if not storage.train_stop_list then
-      storage.train_stop_list = {}
-    end
-    if event.tick % 60 == 0 then
-        log("Persisted train stop count: " .. #storage.train_stop_list)
-    end
-end)
-
-script.on_event(defines.events.on_built_entity,
-  function(event)
-    log("Train stop added.")
-    table.insert(storage.train_stop_list, event.entity)
-  end,
-  {{filter = "name", name = "train-stop"}}
-)
-
-script.on_event(defines.events.on_robot_built_entity,
-  function(event)
-    log("Train stop added.")
-    table.insert(storage.train_stop_list, event.entity)
-  end,
-  {{filter = "name", name = "train-stop"}}
-)
-
 script.on_event(defines.events.on_player_mined_entity,
   function(event)
-    log("Train stop removed.")
-    remove_entity_from_list(event.entity, storage.train_stop_list)
+	storage.dynamicTrainStopSettings[event.entity.unit_number] = nil
   end,
   {{filter = "name", name = "train-stop"}}
 )
 
 script.on_event(defines.events.on_robot_mined_entity,
   function(event)
-    log("Train stop removed.")
-    remove_entity_from_list(event.entity, storage.train_stop_list)
+	storage.dynamicTrainStopSettings[event.entity.unit_number] = nil
   end,
   {{filter = "name", name = "train-stop"}}
 )
 
-script.on_event(defines.events.on_tick,
+script.on_event(defines.events.on_entity_renamed,
   function(event)
-    storage.train_stop_list = storage.train_stop_list or {}
-    for i, train_stop in ipairs(storage.train_stop_list) do
-      local network = train_stop.get_circuit_network(defines.wire_connector_id.circuit_red)
-      if network then
-        local signals = network.signals
-        signals = remove_expected_control_signals(train_stop, signals)
-        set_train_stop_name(train_stop, signals)
+	if event.by_script or event.entity.type ~= 'train-stop' then
+		return
+	end
+	if not storage.dynamicTrainStopSettings[event.entity.unit_number] then 
+		initDynamicTrainStopSettings(event.entity.unit_number, event.entity.backer_name)
+	else 
+		storage.dynamicTrainStopSettings[event.entity.unit_number].name = event.entity.backer_name
+	end
+  end
+)
 
-      end
-    end
+script.on_event(defines.events.on_tick,
+  function(event)  
+    for trainStopUn, trainStopSetting in pairs(storage.dynamicTrainStopSettings) do
```

Comment:
When loading an existing save, the mod crashes because "storage.dynamicTrainStopSettings" is not initialized.

Reply:
pushed a fix